### PR TITLE
Remove support for in-tree build of Zeek plugin/analyzers

### PR DESCRIPTION
**Pull request by rsmmr**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -171,7 +171,6 @@ enable_testing()
 add_subdirectory(hilti)
 add_subdirectory(spicy)
 add_subdirectory(scripts)
```
Comment (bbannier) [inline CMakeLists.txt:173]: Let's also remove the fuzzing-related flags in `./configure` and their handling: `--enable-fuzzing` and `--with-clang-fuzzer-no-main`, and their CMake counterparts `USE_FUZZING` and `CLANG_FUZZER_NO_MAIN`.

File: hilti/toolchain/bin/hiltic.cc [hunk 2]
```diff
@@ -2,8 +2,10 @@
 
 #include <hilti/hilti.h>
 
+#include "base/util.h"
```
Comment (bbannier) [inline hilti/toolchain/bin/hiltic.cc:5]: Include style.

File: .cirrus.yml [hunk 3]
```diff
@@ -4,12 +4,12 @@ environment:
     # Enforce sequential JIT'ing of files for controlled memory usage.
     HILTI_JIT_SEQUENTIAL: 1
 
-clang9_zeek_ubuntu_debug_task:
+    # Branch to use for spicy-plugin and spicy-analyzers tests.
+    SPICY_PLUGIN_BRANCH: topic/robin/plugin-reorg
```
Comment (bbannier) [inline .cirrus.yml:8]: This needs to be adjusted before merging.

File: .cirrus.yml [hunk 4]
```diff
@@ -894,14 +870,55 @@ freebsd12_task:
...
-    - SPICY_INSTALLATION_DIRECTORY=/opt/spicy make -C tests test-all-install
+    - SPICY_INSTALLATION_DIRECTORY=/opt/spicy make -C tests test
   packaging_script:
     - ninja -C build package
 
   packages_artifacts:
     path: build/spicy*.tar.gz
     type: application/gzip
+
+# zeek_plugin_task:
```
Comment (bbannier) [inline .cirrus.yml:885]: Let's remove this.

File: NEWS.rst [hunk 5]
```diff
@@ -31,6 +31,11 @@ Version 1.2
   We previously silently ignored unsupported attributes; now errors are
   raised.
 
+- Remove configure option ``--build-zeek-plugin``. Spicy no longer
+  supports building the Zeek plugin/analyzers in-tree. This used to be
+  available primarily for development purposes, but become challenging
```
Comment (bbannier) [inline NEWS.rst:36]: s/become/became/

File: .cirrus.yml (hunk 4 above)
Comment (rsmmr) [reply .cirrus.yml]: It was just a draft, that's why it was commented out. Revised and activated now.

**Pull request merged**